javalib/src/main/scala/java/net/DatagramPacket.scala

```diff
@@ -0,0 +1,89 @@
+package java.net
+
+final class DatagramPacket(
+    var buf: Array[Byte],
+    var offset: Int,
+    var length: Int,
+    // bufLength: Int,
+    var address: InetAddress,
+    var port: Int
+) {
+  if (length < 0 || offset < 0 || (length + offset) < 0 || ((length + offset) > buf.length)) {
```

Comment:
If both length and offset are non-zero how can (length + offset) ever be < 0.  What am I missing?

Reply:
I think a check that the port is within range is missing here (and possibly elsewhere).

I am suspicious of using -1 for a port. Probably is intended to mean "unspecified". I would have to trace code in more detailed.

It is certainly OK for javalib code to use java `Optional`.
I believe it is OK for it to use Scala Option. (There is
a drive to get rid of Scalalib dependencies in javalib in order to remove some circularity. I think that Scala.js
had to add some compiler support for Option in order
for it to be usable and not considered a Library class.
That discussion is up in the air for SN. I tend to use 
Optional (when I remember) so that I do not get sucked
into that whirlpool.)